docs/modules/generate.md

```diff
@@ -0,0 +1,163 @@
+    - **`$generate.x bs.data`**: The X coordinate of the current step.
+    - **`$generate.y bs.data`**: The Y coordinate of the current step.
+  :::
+```
+
+::::
+::::{tab-item} Fractal shape 2D
+
+```{function} #bs.generate:fractal_shape_2d
+
+Generates a shape in 2D space using a Fractal noise algorithm. The shape is generated at the specified origin, with the callback executed at each step to facilitate custom behavior during the generation process.
+
+:Inputs:
+  **Execution `at <entity>` or `positioned <x> <y> <z> rotated <rot>`**: Origin of the shape.
+
+  **Function macro**:
+  :::{treeview}
+  - {nbt}`compound` Arguments
+    - {nbt}`compound` **run**: Callback to run at each step.
+    - {nbt}`compound` **width**: Width of the shape to generate.
+    - {nbt}`compound` **height**: Height of the shape to generate.
+    - {nbt}`compound` **with**: Shape settings.
+      - {nbt}`string` **direction**: Plane used to generate the shape (`xz`, `xy` or `zy`, default: `xz`).
+      - {nbt}`int` **spacing**: Distance between blocks in the generated shape (default: 1).
+      - {nbt}`int` **limit**: Limit how many steps are executed in a single tick (default: 256).
+      - {nbt}`int` **size**: Granularity of the noise. Lower values increase detail (default: 32).
+      - {nbt}`int` **seed**:  Seed for the noise generation, allowing for reproducibility (default: random).
+      - {nbt}`int` **octaves**: Number of noise layers; more octaves enhance detail (default: 2).
+      - {nbt}`number` **persistence**: Contribution of each octave. Higher means more detail (default: 0.5).
+      - {nbt}`number` **lacunarity**: Frequency increase for each octave. Higher means more rapid frequency increase (default: 2.0).
+  :::
+
+:Outputs:
+  **State**: The callback gets executed at the appropriate coordinates in the world.
+  :::{treeview}
+  - For each step, the following scores are set:
+    - **`$random.fractal_noise_2d bs.out`**: The noise value in the range [-1000, 1000].
+    - **`$generate.x bs.data`**: The X coordinate of the current step.
+    - **`$generate.y bs.data`**: The Y coordinate of the current step.
+  :::
+```
+
+::::
+:::::
+
+*Generate a 3D terrain using a heightmap. This function employs a linear approach; for more interesting terrain, consider scaling the noise values differently according to various ranges. Check out [this video](https://www.youtube.com/watch?v=CSa5O6knuwI) for more insights on terrain generation:*
+
+```mcfunction
+# Once
+function #bs.generate:fractal_shape_2d {width:64,height:64,run:"function bs.generate:shape_2d/demo/heightmap/heightmap",with:{}}
+
+# bs.generate:shape_2d/demo/heightmap/heightmaps
+execute store result storage bs:ctx y int .01 run scoreboard players add $random.fractal_noise_2d bs.out 1000
+function bs.generate:shape_2d/demo/heightmap/run with storage bs:ctx
+
+# bs.generate:shape_2d/demo/heightmap/run
+$fill ~ ~ ~ ~ ~$(y) ~ stone
+```
```

Comment:
For most of the other features, we have an example per tab, not an example for the whole group.

Reply:
This was a deliberate choice since:
- Im lazy
- The only example we have use the same function, and all these functions behave the exact same. I think it's better for now, then if we have more examples that are actually useful, we can split those across tabs